### PR TITLE
Add CORS support for the API requests

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/2674

### CODE REVIEW

**Comment by sumerjabri** [comment]
@jdrossl please also send a PR for `studio-config-override.yaml` in `craftercms` to allow the user to override/make changes.

**Comment by jdrossl** [comment]
@sumerjabri I don't think we need to add them there to allow overriding, any property can be added if needed

File: src/main/resources/crafter/studio/studio-config.yaml [hunk 1]
```diff
@@ -519,4 +519,14 @@ studio.clustering.threadPool.namePrefix: Studio_Clustering_Thread_Pool
...
 studio.clustering.threadPool.corePoolSize: 10
 # Thread pool max size
-studio.clustering.threadPool.maxPoolSize: 20
\ No newline at end of file
+studio.clustering.threadPool.maxPoolSize: 20
+
+################################################################
+##                             CORS                           ##
+################################################################
+studio.cors.disable: false
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/studio-config.yaml:531]: Please document these fields (one comment per field above the field).

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please also add these to the override file in `craftercms` with the comments that you're going to add here. Make sure it's commented out in the override.

The point of the override is to allow some orgs to make updates to tighten things up if required.

**Review by avasquez614 — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**